src/Sulu/Bundle/ContentBundle/Tests/Functional/Controller/NodeControllerTest.php

```diff
@@ -1544,7 +1581,59 @@ public function testCopyNonExistingDestination()
             'POST',
             '/api/nodes/' . $data[0]['id'] . '?webspace=sulu_io&language=en&action=copy&destination=123-123'
         );
-        $this->assertHttpStatusCode(400, $client->getResponse());
+        $this->assertHttpStatusCode(404, $client->getResponse());
+    }
+
+    public function testCopyWithShadow()
+    {
+        $document = $this->createPageDocument();
+        $document->setTitle('test_en');
+        $document->setResourceSegment('/test_en');
+        $document->setStructureType('default');
+        $document->getStructure()->bind([
+            'tags' => [
+                'tag1',
+                'tag2',
+            ],
+            'article' => 'Test English',
+        ]);
+        $this->documentManager->persist($document, 'en', ['parent_path' => '/cmf/sulu_io/contents']);
+        $this->documentManager->flush();
+
+        $document->setTitle('test_de');
+        $document->setResourceSegment('/test_de');
+        $document->setStructureType('default');
+        $document->getStructure()->bind([
+            'tags' => [
+                'tag1',
+                'tag2',
+            ],
+            'article' => 'Test German',
+        ]);
+        $this->documentManager->persist($document, 'de', ['parent_path' => '/cmf/sulu_io/contents']);
+        $this->documentManager->flush();
+
+        $document = $this->documentManager->find($document->getUuid(), 'de');
+        $document->setShadowLocaleEnabled(true);
+        $document->setShadowLocale('en');
+        $this->documentManager->persist($document, 'de');
+        $this->documentManager->flush();
+
+        $client = $this->createAuthenticatedClient();
+
+        $client->request(
+            'POST',
+            '/api/nodes/' . $document->getUuid()
```

Comment:
Tried it, but looks a bit weird to me, since `%` is a valid character in URLs...
```php
sprintf(
                '/api/nodes/%s?webspace=sulu_io&language=en&action=copy&destination=%s',
                $document->getUuid(),
                $document->getUuid()
            )
```